### DISCUSSION TITLE
Eager loading users with roles and groups  in the control panel (eloquent repository)

### DISCUSSION THREAD

**Discussion by flyingeek** [statamic/cms#6392] [Q&A] [answered]
Hello, I've follow the [statamic.dev/tips](https://statamic.dev/tips/storing-users-in-a-database) to store users in in database. It works fine but when the cp shows users, it performs one request per user displayed to load the role and another one to load the group.

I'm new to Statamic/Laravel but I tried to add in App\Models\User:
```php
protected $with = ['roles', 'groups']
```
But it did not help. Is there a way to reduce the amount of SQL requests ?

Bonus question: is there a way to generate automagically a migration when I change the user blueprint ?

**Comment by ryanmitchell** [comment, marked as answer]
@flyingeek there are some performance improvements in this PR which should reduce the number of request: https://github.com/statamic/cms/pull/5686

> Bonus question: is there a way to generate automagically a migration when I change the user blueprint ?

Unfortunately not!
